dotCMS/src/main/java/com/dotmarketing/portlets/contentlet/business/HostAPIImpl.java

```diff
@@ -1019,4 +1020,42 @@ public void updateMenuLinks(Host workinghost,Host updatedhost) throws DotDataExc
 		return hosts;
 
 	}
+	
+	public PaginatedArrayList<Host> search(String filter, boolean showArchived, boolean showSystemHost, int limit, int offset, User user, boolean respectFrontendRoles) throws DotDataException, DotSecurityException {
+		try {
+			Structure st = CacheLocator.getContentTypeCache().getStructureByVelocityVarName("Host");
+			String condition="";
+			
+			if(showArchived){
+				condition=" +(working:true deleted:true)";
+			}else {
+				condition=" +working:true";
+			}
+			
+			if(UtilMethods.isSet(filter)){
+				condition += " +Host.hostName:"+filter.trim()+"*";
+			}
+			if(!showSystemHost){
+				condition += " +Host.isSystemHost:false";
+			}
+			PaginatedArrayList<Contentlet> list = (PaginatedArrayList<Contentlet>)APILocator.getContentletAPI().search("+structureInode:" + st.getInode() + condition, limit, offset, "Host.hostName", user, respectFrontendRoles);
+			
+			return convertToHostPaginatedArrayList(list);
+		} catch (Exception e) {
+			Logger.error(HostAPIImpl.class, e.getMessage(), e);
+			throw new DotRuntimeException(e.getMessage(), e);
+		}
+	}
+	
+	private PaginatedArrayList<Host> convertToHostPaginatedArrayList(PaginatedArrayList<Contentlet> list) {

```

Comment:
may be this temp collection is not necessary, you may want to use stream().map ( contentlet -> new Host(contentlet)) instead